codegen/module.go

```diff
@@ -1153,6 +1153,12 @@ func (system *ModuleSystem) IncrementalBuild(
 		}
 	}
 
+	for i, hook := range system.postGenHook {
+		if err := hook(resolvedModules); err != nil {
```

Comment:
I think you want to `toBeBuildModules` so that postGenHook is run only on those for incrementalBuild.